### PR TITLE
fix: log correct reasons for including a polyfill

### LINKED ISSUE #110: [Bug]: preset-env with "debug: true" lists every target rather than the one which caused the polyfill
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

@babel/cli

### Input code

```js
// code doesn't really matter, it just needs to cause some polyfills
[...document.querySelectorAll('.something')].forEach(
    (el) => el.classList.remove('something')
);
```

### Configuration file name

babel.config.json

### Configuration

```json
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "debug": true,
        "targets": [
          "> 0.25%",
          "not dead",
          "not IE 11"
        ],
        "useBuiltIns": "usage",
        "corejs": "3.6.5"
      }
    ]
  ]
}
```

### Current behavior (started in 7.13.0):
preset-env with "debug: true" lists every target next to every applied polyfill, rather than only the target which caused the polyfill.

Example current behavior:
```
[babel-polyfill-test/src/test.js]
The corejs3 polyfill added the following polyfills:
  es.object.to-string { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
  web.dom-collections.for-each { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
  es.symbol { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
  es.symbol.description { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
  es.symbol.iterator { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
  es.array.iterator { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
  es.string.iterator { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
  web.dom-collections.iterator { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
  es.array.from { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
  es.array.slice { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
  es.regexp.exec { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
  es.regexp.test { "android":"4.4.3", "chrome":"92", "edge":"96", "firefox":"94", "ios":"12.2", "opera":"81", "safari":"13.1", "samsung":"15" }
```

### Expected behavior (last working version was 7.12.17):
Show the specific targets which caused each polyfill.

Example expected behavior, which is how it worked up to and including 7.12.17:
```
[babel-polyfill-test/src/test.js] Added following core-js polyfill:
  web.dom-collections.for-each { "android":"4.4.3" }
```

### Environment

@babel/preset-env@7.16.7
node v16.13.1
npm 8.1.2
Ubuntu 20.04

### Possible solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dreyks**
closes #110

### CODE REVIEW

**Comment by dreyks** [comment]
force-pushed to fix a linter issue

File: packages/babel-helper-define-polyfill-provider/src/index.ts [hunk 1]
```diff
@@ -220,10 +220,9 @@ function instantiateProvider<Options>(
       if (!debug || !name) return;
 
       if (debugLog().polyfills.has(provider.name)) return;
-      debugLog().polyfills.set(
-        name,
-        polyfillsSupport && name && polyfillsSupport[name],
-      );
+      debugLog().polyfills.add(name);
+      if (!debugLog().polyfillsSupport)
+        debugLog().polyfillsSupport = polyfillsSupport;
```
Comment (JLHwung) [inline packages/babel-helper-define-polyfill-provider/src/index.ts:225]:
```suggestion
      debugLog().polyfillsSupport ??= polyfillsSupport;
```

**Review by JLHwung — APPROVED** [review]
Thanks!

File: packages/babel-helper-define-polyfill-provider/src/index.ts [hunk 2]
```diff
@@ -375,9 +374,13 @@ export default function definePolyfillProvider<Options>(
           );
         }
 
-        for (const [name, support] of debugLog.polyfills) {
-          if (support) {
-            const filteredTargets = getInclusionReasons(name, targets, support);
+        for (const name of debugLog.polyfills) {
+          if (debugLog.polyfillsSupport && debugLog.polyfillsSupport[name]) {
```
Comment (nicolo-ribaudo) [inline packages/babel-helper-define-polyfill-provider/src/index.ts:378]:
```suggestion
          if (debugLog.polyfillsSupport?.polyfillsSupport[name]) {
```

**Pull request merged**